pkg/apis/monitoring/v1beta1/alertmanager_config_types.go

```diff
@@ -1291,6 +1295,156 @@ type RocketChatActionConfig struct {
+	// +optional
+	IconEmoji *string `json:"iconEmoji,omitempty"`
+	// attachments define the message attachments used for richer formatting options.
+	// It is for compatibility with Slack.
+	// +optional
+	Attachments []MattermostAttachmentConfig `json:"attachments,omitempty"`
+	// props defines the extra information to be sent to Mattermost.
+	// +optional
+	Props *MattermostPropsConfig `json:"props,omitempty"`
+	// priority defines the priority parameters of the message.
+	// +optional
+	Priority *MattermostPriorityConfig `json:"priority,omitempty"`
+	// httpConfig defines the HTTP client configuration for Mattermost webhook requests.
+	// +optional
+	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
+}
+
+// MattermostAttachmentConfig configures attachment parameters for the notifications.
+// For more information, please refer to the Mattermost document:
+// https://developers.mattermost.com/integrate/reference/message-attachments/
+type MattermostAttachmentConfig struct {
+	// fallback defines a required plain-text summary of the attachment.
+	// This is used in notifications, and in clients that don’t support formatted text (e.g. IRC).
+	// +optional
+	Fallback *string `json:"fallback,omitempty"`
+	// color defines a hex color code that will be used as
+	// the left border color for the attachment.
+	// If not specified, it will default to match the channel sidebar header background color.
+	// +optional
+	Color *string `json:"color,omitempty"`
+	// pretext defines an optional line of text that will be shown above the attachment. Supports @mentions.
+	// +optional
+	Pretext *string `json:"pretext,omitempty"`
+	// text defines the text to be included in the attachment.
+	// It can be formatted using Markdown.
+	// For long texts, the message is collapsed and a “Show More” link is added to expand the message. Supports @mentions.
+	// +optional
+	Text *string `json:"text,omitempty"`
+	// authorName defines an optional name used to identify the author.
+	// It will be included in a small section at the top of the attachment.
+	// +optional
+	AuthorName *string `json:"authorName,omitempty"`
+	// authorLink defines optional URL used to hyperlink the author_name.
+	// If no author_name is specified, this field does nothing.
+	// +optional
+	AuthorLink *URL `json:"authorLink,omitempty"`
+	// authorIcon defines an optional URL used to display a 16x16 pixel icon beside the author's name.
+	// +optional
+	AuthorIcon *URL `json:"authorIcon,omitempty"`
+	// title defines an optional title displayed below the author information in the attachment.
+	// +optional
+	Title *string `json:"title,omitempty"`
+	// titleLink defines an optional URL used to hyperlink the title.
+	// If no title is specified, this field does nothing.
+	// +optional
+	TitleLink *URL `json:"titleLink,omitempty"`
+	// fields defines a list of Slack fields that are sent with each notification.
+	// +optional
+	Fields []SlackField `json:"fields,omitempty"`
```

Comment:
this should be a new type MattermostField struct, pls check for reference https://github.com/UndeadDemidov/alertmanager/blob/main/config/notifiers.go#L1098